### DISCUSSION TITLE
Omnet Federate missing file: rebuild_federate.sh

### DISCUSSION THREAD

**Discussion by LDVictor** [eclipse/mosaic#93] [Q&A]
Hello all,

I'm trying to rebuild Omnet Federate following this tutorial: https://www.eclipse.org/mosaic/docs/extending_mosaic/omnetpp_details/.

But when I need to specify the location of rebuild, the indicated path in the tutorial is "federate/rebuild_federate.sh", but this file doesn't exist in Federate directory. I searched in all of Omnet Federate folders, but I didn't find. Anybody knows when I can find this file?

**Comment by DemingCheng** [comment]
I also meet this problem，have you solved it?

**Reply by DemingCheng** [reply]
@kschrab Could you please offer the rebuild_federate.sh? I'm a rookie in the eclipse mosaic and omnetpp so I would really appreciate any help from you.

Also, is it possible to combine the aodv in omnetpp with the eclipse mosaic?

Thanks in advance.

**Reply by LDVictor** [reply]
I solved this problem making a new build of Omnet Federate changing the files in ZIP with my modifications. The problem is: when I do another modification, I need to build again copying the new modifications in the files inside ZIP.

**Comment by kschrab** [comment]
Hi, the documentation of the federate building process is indeed outdated. However, I've found an old version of `rebuild_federate.sh` script:

```sh
#!/usr/bin/env bash

(cd .. && make config=debug clean && make config=debug)

if [ ! -f "./federate_dbg" ] && [ -f "../bin/Debug/omnetpp-federate" ]; then
   ln -s "../bin/Debug/omnetpp-federate" "federate_dbg"
fi
if [ ! -f "./federate" ] && [ -f "../bin/Debug/omnetpp-federate" ]; then
   ln -s "../bin/Debug/omnetpp-federate" "federate"
fi
if [ ! -f "./libINET_dbg.so" ]; then
   ln -s "../../inet_src/src/libINET_dbg.so" "libINET_dbg.so"
fi
```
I do not know if this script still works. Right now we are working on improving the building process and documentation, but this still will take a bit of time. If you find any helpful advice for the community (e.g. if this script actually works) please spread it :)
